evolver/controller/standard/tests/test_chemostat.py

```diff
@@ -18,7 +19,9 @@ def add_mock_hardware(evolver):
     def setup_hw_mock(mock, cls):
         mock.inputs = []
         mock.Input.side_effect = lambda **a: cls.Input(**a)
-        mock.set.side_effect = lambda a: mock.inputs.append(a)
+        mock.set.side_effect = lambda *args, **kwargs: mock.inputs.append(
+            EffectorDriver._get_input_from_args(cls, *args, **kwargs)
```

Comment:
this supports inputs in the same way as super's set method (either via Input object or args of Input class)